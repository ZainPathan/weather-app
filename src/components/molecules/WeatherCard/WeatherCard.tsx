import React from 'react';
import {
    Card,
    CardContent,
    Typography
} from '@material-ui/core';
import clsx from 'clsx';
import useWeatherCardStyles from './styles/WeatherCard.style';

interface WeatherCardProps {
    currentCardIndex: number,
    weatherDate: string,
    temperatureDetails: string,
    index: number
}

const WeatherCard = ({currentCardIndex, weatherDate, temperatureDetails, index}: WeatherCardProps) => {
    const classes = useWeatherCardStyles();

    return (
        <Card
            variant="outlined"
            className={
                (currentCardIndex === index ) ?
                    clsx(classes.weatherCard, classes.selectedWeatherCard) :
                    classes.weatherCard
            }
        >
            <CardContent>
                <Typography
                    variant="h5"
                    component="h2"
                >
                    Temperature:&nbsp;
                    {temperatureDetails}
                </Typography>
                <br/>
                <Typography
                    variant="h5"
                    component="h2"
                >
                    Date: {weatherDate}
                </Typography>
            </CardContent>
        </Card>
    )
}

export default WeatherCard;
// export {};