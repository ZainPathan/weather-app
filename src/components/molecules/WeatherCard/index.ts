import WeatherCard from './WeatherCard';

export {
    WeatherCard
};

export default WeatherCard;