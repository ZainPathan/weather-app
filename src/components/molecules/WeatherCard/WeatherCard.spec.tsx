import React from 'react';
import renderer from 'react-test-renderer';
import WeatherCard from './WeatherCard';

describe('WeatherCard component tests', () => {
    const mock = {
        currentCardIndex: 1,
        index: 0,
        temperatureDetails: '68Â°F',
        weatherDate: '18 Jul 21'
    }

    it('renders properly', () => {
        const tree = renderer
            .create(
                <WeatherCard
                    currentCardIndex={mock.currentCardIndex}
                    weatherDate={mock.weatherDate}
                    temperatureDetails={mock.temperatureDetails}
                    index={mock.index}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
