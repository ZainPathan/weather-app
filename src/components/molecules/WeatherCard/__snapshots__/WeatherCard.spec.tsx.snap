// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component tests renders properly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-weatherCard-1 MuiPaper-outlined MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Temperature: 
      68°F
    </h2>
    <br />
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Date: 
      18 Jul 21
    </h2>
  </div>
</div>
`;
